rectools/visuals/metrics_app.py

```diff
@@ -134,13 +165,54 @@ def _validate_input_data(self) -> None:
         if len(self.models_metrics.columns) < 3:
             raise KeyError("`metrics_data` DataFrame assumed to have at least one metric column")
 
+    def _validate_models_metadata(self) -> None:
+        if not isinstance(self.models_metadata, pd.DataFrame):
+            raise ValueError("Incorrect input type. `models_metadata` should be a DataFrame")
+        if Columns.Model not in self.models_metadata.columns:
+            raise KeyError("Missing `Model`` column in `models_metadata` DataFrame")
+        if self.models_metadata[Columns.Model].nunique() != len(self.models_metadata):
+            raise ValueError("Found ambiguous values in `Model` column of `models_metadata`")
+
     @lru_cache
-    def _make_chart_data(self, fold_number: int) -> pd.DataFrame:  # pragma: no cover
-        return self.models_metrics[self.models_metrics[Columns.Split] == fold_number].drop(columns=Columns.Split)
+    def _make_chart_data(self, fold_number: int) -> pd.DataFrame:
+        return (
+            self.models_metrics[self.models_metrics[Columns.Split] == fold_number]
+            .drop(columns=Columns.Split)
+            .merge(self.models_metadata, on=Columns.Model, how="left")
```

Comment:
I have a feeling that it will be more clear to merge `models_metadata` with `models_metrics` one time during `contruct` and also remember col names for metrics and meta. And then accept merged data and col names in __init__. WDYT?
@feldlime WDYT?

Reply:
Keeping in mind that we also want to visualize the data for [the case when we don't have a folds column](https://github.com/MobileTeleSystems/RecTools/pull/154#discussion_r1649685516), idea to make some main table before finalizing sounds nice.
But is it ok to have different arguments for `construct` and `init` methods? Seems a little bit confusing for me. But I can create additional method for data merging and cache the result. Then grab it in three other methods: for a averaged data, for a fold data and a data without any folds.

Reply:
It is definetely ok to have different argument for `construct` and `init`, it is very reasonable and common. So you can merge during `construct` and pass merged data to `init`

Reply:
Done